test/e2e/base_user_test.go

```diff
@@ -139,6 +140,7 @@ func (s *baseUserIntegrationTest) reactivateAndCheckUser(userSignup *v1alpha1.Us
 	userSignup, err = s.hostAwait.UpdateUserSignupSpec(userSignup.Name, func(us *v1alpha1.UserSignup) {
 		states.SetDeactivating(us, false)
 		states.SetDeactivated(us, false)
+		states.SetApproved(us, true)
```

Comment:
Why we need to explicitly set the states to Approved here but we didn't have to do it with the old API?

Reply:
We require this now because the `SetDeactivated()` function sets the `approved` state to false.

Reply:
Technically, we could just call `SetApproved()` and we get both `deactivating` and `deactivated` set to false for us for free.

Reply:
But `states.SetDeactivated(us, false)` does not touch "approved", does it? https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/states/state_manager.go#L52
It deletes it in case of `states.SetDeactivated(us, true)`. But not for `states.SetDeactivated(us, false)`.

Reply:
You're correct, but it's [this line](https://github.com/codeready-toolchain/toolchain-e2e/blob/master/test/e2e/base_user_test.go#L103) that's calling it, not the prior line.

Reply:
I see... So, for manual re-activation/approval we need to restore this state. This should be fine.

Reply:
@sbryzak 
I think that we should not call the `states.SetApproved(us, true)`  here, otherwise, you won't be able to verify reactivation with automatic approval. By setting `states=approved` you approve the UserSignup manually, which is not the purpose of this function!

> We require this now because the SetDeactivated() function sets the approved state to false.

I think that it only deletes the `approved` from the states: https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/states/state_manager.go#L49-L56 in addition, the `SetDeactivated` is called with `false` which means that it won't touch the `approved` state at all.

If I'm not mistaken, when we reactivate (without requiring phone verification), then the `states` should be empty.